### PR TITLE
Add test case verifiying capsule sync in presence of null facet

### PR DESCRIPTION
**Pull request by sambible**
### Problem Statement
Customers have run into issues syncing capsules when hosts have left behind a null subscription facet. 

### Solution
This adds a test case covering this issue.

### Related Issues
https://issues.redhat.com/browse/SAT-40496
and
https://issues.redhat.com/browse/SAT-39794
### PRT test Cases example
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k 'test_capsule_sync_nulled_subscription_facet'

### CODE REVIEW

**Review by ColeHiggins2 — APPROVED** [review]

File: tests/foreman/api/test_capsulecontent.py [hunk 1]
```diff
@@ -2157,6 +2157,92 @@ def test_cleanup_orphaned_content(
...
+        )[0]
+        assert nc.id == host.content_facet_attributes['content_source_id'], (
+            'Expected to see the Capsule as the content source'
+        )
+        # Unregister the contenthost, leaving a null content facet behind.
+        assert rhel_contenthost.unregister().status == 0
+        # Perform a full capsule sync
+        timestamp = datetime.now(UTC)
+        nailgun_capsule = module_capsule_configured.nailgun_capsule
+        sync_status = nailgun_capsule.content_sync(timeout='90m')
```
Comment (vsedmik) [inline tests/foreman/api/test_capsulecontent.py]: To hit the error condition, do we need the full/complete capsule sync or is the optimized one just enough @sambible @sjha4 ?

https://issues.redhat.com/browse/SAT-39794?focusedId=28652994&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-28652994

**Review by vsedmik — COMMENTED** [review]
Looks good to me, I have just one question bellow

**Review by synkd — APPROVED** [review]
ACK pending answer to most recent question
